src/clj/commiteth/routes/services.clj

```diff
@@ -237,4 +257,19 @@
                          :auth-rules authenticated?
                          :current-user user
                          (log/debug "/user/bounties")
-                         (ok (user-bounties user))))))
+                         (ok (user-bounties user)))
+                    (POST "/revoke"  {{issue-id         :issue-id
+                                       contract-address :contract-address
+                                       owner-address    :owner-address} :params}
+                          :auth-rules authenticated?
+                          :current-user user
+                          (do (log/infof "calling revoke-initiate for %s with %s %s" issue-id contract-address owner-address)
+                              (if-let [{:keys [execute-hash execute-write]} (execute-revocation issue-id contract-address owner-address)]
+                                (if (scheduler/poll-transaction-logs execute-hash contract-address)
```

Comment:
Instead of having to synchronously poll here in request handler, can we instead rely on `load-data` fn from `core.cljs` to periodically fetch whether there is a confirm hash for the revocation tx? Fetching confirm hash involves invoking `multisig/find-confirmation-tx-id`, which is very similar thing to what `scheduler/contract-confirmation-logs` does.

Perhaps we could re-use `:load-owner-bounties` event for this purpose (given that `winner-login` field will be set to owner's login so that `owner-bounties` query works as expected)

Reply:
@siphiuel yes this really is the crux of the PR in my opinion, really good to discuss. 

It seems like the main thing is knowing *when* to call `update-confirm-hash` and there is some similarity between  `scheduler/contract-confirmation-logs` and `multisig/find-confirmation-tx-id` because the former is basically ensuring that the conditions will be met for the latter to return something when it is called in this context 😄 

I agree it would be nice to trim out this excess polling, but if I do I'm still not sure what the most succinct thing is to pay attention to for ensuring that the execute_hash is mined? Was thinking the `tx-tracker` you introduced might come in handy here? 

This is kind of a tricky thing to reason about so let me know if I'm missing something obvious, but the reason I haven't relied on `load-data` or `owner-bounties` yet is that if `update-confirm-hash` is not called manually as listed above, it will not be picked up by the scheduler thread since a revoked bounty has no associated PR (one of the conditions of `db/pending-payouts`). 

Thanks again for checking this out, let me know what you think!

Reply:
@siphiuel yes that's a good point. I believe if I created a separate scheduler job to check for receipts of revoked bounties (they won't show up in the current query because there is no associated PR), then it should be possible to just follow the consistent workflow we already have rather than polling from within the request handler. 

I guess my main question then becomes the mechanism by which we would know to either call `load-data` or `owner-bounties` to get the confirm-hash in hand in order to call confirm-payout. Do we have a way to call these handlers periodically to the extent they can't be event driven? (happy to write such a feature – recurring dispatch or something – too, just curious if it exists already for the use case you describe?)

Thanks!